spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```
@@ -750,6 +752,9 @@ class DistriOptimizer[T: ClassTag] (
              */
             if (System.nanoTime() - lastFailureTimestamp < maxRetry * retryTimeInterval * 1e9) {
               retryNum += 1
+              if (retryNum == maxRetry) {
```

What if ```  (checkpointPath.isDefined)  ``` is false?  exception would be swallow silently.